lib/ts/xmlhttprequest.ts

```diff
@@ -307,10 +303,7 @@ export function addInterceptorsToXMLHttpRequest() {
                 });
                 // The original version "combines" headers according to MDN.
```

Comment:
please add a test for this PR?

Also, i won't merge this PR until after our call with that user.